cloudpathlib/cloudpath.py

```diff
@@ -595,6 +595,44 @@ def rmtree(self):
             raise ValueError(f"Path {self} is a file; call unlink instead of rmtree.")
         self.client._remove(self)
 
+    def upload_from(self, source: Union[str, os.PathLike]):
+        """ Upload a single file to the cloud path. """
+        source = Path(source)
+        if not source.is_file():
+            raise ValueError("Method upload_from implemented only when the source is a file.")
+        if self.is_dir():
+            self.client._upload_file(local_path=self, cloud_path=self / source.name)
+        else:
+            self.client._upload_file(local_path=self, cloud_path=self)
+
+    def copy(self, destination: "CloudPath"):
+        """ Copy self to destination folder of file, if self is a file. """
+        if not self.is_file():
+            raise ValueError(
+                f"Path {self} should be a file. To copy a directory tree use the method copytree."
+            )
+        temp_file = self._local
+        temp_file.parent.mkdir(parents=True)
+        self.download_to(temp_file)
```

Comment:
Instead of download to a `temp_file`, you can do `self.fspath` which will use the cached version if it is downloaded and download it to the cache if not. So the lines below can just use that. For example `destination.upload_from(self.fspath)`